app/src/main/java/pl/futurecollars/invoicing/db/FileBasedDatabase.java

```diff
@@ -0,0 +1,113 @@
+package pl.futurecollars.invoicing.db;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import pl.futurecollars.invoicing.config.Configuration;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.service.IdService;
+import pl.futurecollars.invoicing.service.JsonService;
+
+public class FileBasedDatabase implements Database {
+
+    private final File invoicesFile;
+    private final Path invoicesPath;
+    private final IdService idService;
+    private final JsonService jsonService = new JsonService();
+
+    public FileBasedDatabase() {
+        invoicesFile = new File(Configuration.INVOICES_PATH);
+        invoicesPath = Path.of(Configuration.INVOICES_PATH);
+        idService = new IdService(Path.of(Configuration.ID_PATH));
+    }
+
+    public FileBasedDatabase(File invoicesFile, Path invoicesPath, IdService idService) {
+        this.invoicesFile = invoicesFile;
+        this.invoicesPath = invoicesPath;
+        this.idService = idService;
+    }
+
+    @Override
+    public int save(Invoice invoice) {
+        int id = idService.getId();
+        invoice.setId(id);
+        idService.setId();
+
+        try {
+            Files.writeString(invoicesPath, jsonService.objectToString(invoice),
+                    invoicesFile.exists() ? StandardOpenOption.APPEND : StandardOpenOption.CREATE_NEW);
+            return id;
+        } catch (IOException exception) {
+            exception.printStackTrace();
+        }
+
+        return -1;
```

Comment:
tutaj zamiast zwracać -1 lepiej od razu w catch byłoby wyrzucić jakiś RuntimeException (albo własny wyjątek, który go rozszerza) - problem z takimi magic numberami jest taki, że jest zwracany w ten sam sposób jak prawidłowy identyfikator i trudniej to wykryć jako błąd

Reply:
done